pkg/operator/ceph/csi/ceph_connection.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2024 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package csi
+
+import (
+	"github.com/pkg/errors"
+
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	cephclient "github.com/rook/rook/pkg/daemon/ceph/client"
+
+	csiopv1a1 "github.com/ceph/ceph-csi-operator/api/v1alpha1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func CreateUpdateCephConnection(c client.Client, clusterInfo *cephclient.ClusterInfo, clusterSpec cephv1.ClusterSpec) error {
+
+	logger.Infof("Configuring ceph connection CR %q in namespace %q", clusterInfo.NamespacedName().Name, clusterInfo.NamespacedName().Namespace)
+	csiCephConnection := &csiopv1a1.CephConnection{}
+
+	csiCephConnection.Name = clusterInfo.NamespacedName().Name
+	csiCephConnection.Namespace = clusterInfo.NamespacedName().Namespace
+
+	spec, err := generateCephConnSpec(c, clusterInfo, csiCephConnection.Spec, clusterSpec)
+	if err != nil {
+		return errors.Wrapf(err, "failed to set ceph connection CR %s in namespace %s", csiCephConnection.Name, clusterInfo.Namespace)
+	}
+
+	err = clusterInfo.OwnerInfo.SetOwnerReference(csiCephConnection)
+	if err != nil {
+		return errors.Wrapf(err, "failed to set owner reference for ceph connection CR %q", csiCephConnection.Name)
+	}
+
+	err = c.Get(clusterInfo.Context, types.NamespacedName{Name: csiCephConnection.Name, Namespace: csiCephConnection.Namespace}, csiCephConnection)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			csiCephConnection.Spec = spec
+			err = c.Create(clusterInfo.Context, csiCephConnection)
+			if err != nil {
+				return errors.Wrapf(err, "failed to create ceph connection CR %s", csiCephConnection.Name)
```

Comment:
```suggestion
				return errors.Wrapf(err, "failed to create ceph connection CR %q", csiCephConnection.Name)
```

Make it more consistent across all the log messages. Currently its %s at some places and %q at others.